### PR TITLE
Remove __author__ lines, add contributors.md

### LINKED ISSUE #626: Remove __author__ from all files
Might be worth putting a section in `CONTRIBUTING` giving a shout to all the author's over time?

### PR DESCRIPTION
**Pull request by theacodes**
Resolves #626

### CODE REVIEW

**Comment by jonparrott** [comment]
I knew our author lines were inaccurate at best, but they were vastly underrepresenting the people who have contributed to this library.

**Comment by dhermes** [comment]
Insane number of contributors!

Do we want a special section for past maintainers like Joe and Craig?

**Comment by dhermes** [comment]
LGTM save for my comment.

We should wait for @nathanielmanistaatgoogle to weigh in

File: oauth2client/_helpers.py [hunk 1]
```diff
@@ -26,11 +26,6 @@
 from six.moves import urllib
 
 
-__author__ = (
-    'rafek@google.com (Rafe Kaplan)',
-    'guido@google.com (Guido van Rossum)',
-)
-
```
Comment (waprin) [inline oauth2client/_helpers.py:33]: These guys are getting removed from here but aren't in contributors section, not sure if intentional?

**Comment by jonparrott** [comment]
Happy to add a previous maintainers section if both you and @nathanielmanistaatgoogle  think it's important. :)

File: oauth2client/_helpers.py (hunk 1 above)
Comment (theacodes) [inline oauth2client/_helpers.py:33]: neither have actual commits in the repo. Looking into the history, it's likely that the very first version of this file was copied from elsewhere in Google and guido wrote that file.

Comment (dhermes) [inline oauth2client/_helpers.py:33]: They came from `protorpc.util`, where `@positional` was cribbed from

Comment (waprin) [inline oauth2client/_helpers.py:33]: Ok, if code they wrote is used in the repo seems like a very small and easy bone to throw them to put them in the contributors file. Whoever this Guido guy is could probably use the acknowledgement and recognition.

**Comment by pferate** [comment]
I agree with @dhermes on having a separate section for former maintainers.

File: oauth2client/_helpers.py (hunk 1 above)
Comment (theacodes) [inline oauth2client/_helpers.py:33]: Done.

**Comment by jonparrott** [comment]
Former maintainers section added. @nathanielmanistaatgoogle PTAL?

**Pull request merged**